### ISSUE TITLE
trigger-sim broken for frame objects that are not in simclasses or dataclasses (Trac #1695)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1687]
trigger-sim broken for frame objects that are not in simclasses or dataclasses. Found this because trunk trigger-sim does not work for GENIE simulation. GENIE puts the `I3GENIEResultsDict` in the frame. When trigger-sim loops through the frame objects to find the `I3TriggerHierarchy`, python will hit the `I3GENIEResultsDict` in the frame and will throw an `Unregistred Class` error because the  `I3GENIEResultsDict` is not part of dataclasses or simclasses.

It is a bad idea to assume that everything in a frame is deserializable. The correct solution is to pass a list of `I3TriggerHierarchy`s and have a default that covers the standard use case. 

Migrated from https://code.icecube.wisc.edu/ticket/1695
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:11:45", 
    "description": "trigger-sim broken for frame objects that are not in simclasses or dataclasses. Found this because trunk trigger-sim does not work for GENIE simulation. GENIE puts the `I3GENIEResultsDict` in the frame. When trigger-sim loops through the frame objects to find the `I3TriggerHierarchy`, python will hit the `I3GENIEResultsDict` in the frame and will throw an `Unregistred Class` error because the  `I3GENIEResultsDict` is not part of dataclasses or simclasses.\n\nIt is a bad idea to assume that everything in a frame is deserializable. The correct solution is to pass a list of `I3TriggerHierarchy`s and have a default that covers the standard use case. ", 
    "reporter": "benedikt.riedel", 
    "cc": "kclark, melanie.day", 
    "resolution": "fixed", 
    "_ts": "1550067105393059", 
    "component": "combo simulation", 
    "summary": "trigger-sim broken for frame objects that are not in simclasses or dataclasses", 
    "priority": "blocker", 
    "keywords": "", 
    "time": "2016-05-06T21:42:14", 
    "milestone": "", 
    "owner": "olivas", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160506T21:43:53: olivas commented:

> Can you paste the error?

**Comment by icecube-trac** [comment]
Trac update at 20160506T21:45:42: 
* benedikt.riedel commented:

> ```File "/home/kenclark/sim_trunk/build/lib/icecube/trigger_sim/modules/time_shifter.py", line 58, in DAQ
>    for key,f in frame.items():
> RuntimeError: Frame caught exception "unregistered class" for key "I3GENIEResultDict" of type```


* benedikt.riedel changed _comment0 from "`RuntimeError: Frame caught exception "unregistered class" for key "I3GENIEResultDict"....`" to "1462571221385582"
* benedikt.riedel changed _comment1 from:

> `File "/home/kenclark/sim_trunk/build/lib/icecube/trigger_sim/modules/time_shifter.py", line 58, in DAQ
>    for key,f in frame.items():
> RuntimeError: Frame caught exception "unregistered class" for key "I3GENIEResultDict" of type`

to:

> 1462571228923580

**Comment by icecube-trac** [comment]
Trac update at 20160506T22:46:06: 
* olivas changed status from "new" to "closed"
* olivas changed resolution from "" to "fixed"
* olivas commented:

> In changeset 145572:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="145572"
> effectively 'skip unregistered classes' in time_shifter.  closes #1687.
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:11:45: 
* olivas changed milestone from "IceSim 5.0" to ""
* olivas commented:

> Milestone IceSim 5.0 deleted
